docroot/sites/all/modules/custom/node_stat/node_stat.module

```diff
@@ -0,0 +1,78 @@
+<?php
+
+function node_stat_node_view($node, $view_mode = 'full') {
+  if ($view_mode == 'full') {
+
+    global $user;
+
+    $record = node_stat_get_record($node->nid, $user->uid);
+
+    if ($record === FALSE) {
+      $record = new StdClass();
+      $record->nid = $node->nid;
+      $record->uid = $user->uid;
+      $record->last_viewed = time();
```

Comment:
You probably don't need this `$record->last_viewed = time();` executed conditionally if you do the same after the condition (outside) immediately. It will do the same job twice.